src/transformers/generation_utils.py

```diff
@@ -47,71 +70,109 @@ def adjust_logits_during_generation(self, logits, **kwargs):
         """
         return logits
 
-    def enforce_repetition_penalty_(self, lprobs, batch_size, num_beams, prev_output_tokens, repetition_penalty):
+    def _prepare_encoder_decoder_kwargs(self, input_ids, model_kwargs):
+        # retrieve encoder hidden states
+        encoder = self.get_encoder()
+        encoder_kwargs = {
+            argument: value for argument, value in model_kwargs.items() if not argument.startswith("decoder_")
+        }
+        model_kwargs["encoder_outputs"]: ModelOutput = encoder(input_ids, return_dict=True, **encoder_kwargs)
+        return model_kwargs
+
+    def _prepare_decoder_input_ids(self, input_ids, decoder_start_token_id=None, bos_token_id=None, **model_kwargs):
+
+        if "decoder_input_ids" in model_kwargs:
+            return model_kwargs["decoder_input_ids"]
+
+        decoder_start_token_id = self._get_decoder_start_token_id(decoder_start_token_id, bos_token_id)
+        decoder_input_ids = (
+            torch.ones((input_ids.shape[0], 1), dtype=input_ids.dtype, device=input_ids.device)
+            * decoder_start_token_id
+        )
+        return decoder_input_ids
+
+    def _get_decoder_start_token_id(self, decoder_start_token_id, bos_token_id):
```

Comment:
Yeah I see what you mean. I guess this would apply in 99% of the cases, but a problem could arrive if a user has defined `config.decoder_start_token_id` differently than `config.decoder.decoder_start_token_id` => this would then be overwritten at init.